notes/lecture17.tex

```diff
@@ -0,0 +1,189 @@
+\end{definition}
+
+\begin{definition}[Global minimum]
+A point $x^*$ is an unconstrained \textit{global minimum} if $f(x^*) \le f(x)$ for all $x$.
+\end{definition}
+
+For both definitions, we say ''strict'' if these inequalities are strict.
+
+\begin{proposition}[Necessary Conditions for local minimum]
+\propositionlabel{nc for local min}
+Let $x^*$ be an unconstrained local minimum of $f\colon \R^n \to\R$ and assume $f$ is continuously differentiable ($C^1$) in an open set containing $x^*$. Then
+\begin{enumerate}
+    \item $\nabla f(x^*) = 0$ (First-Order Necessary Condition)
+    \item If in addition $f$ is twice continuously differentiable in an open set around $x^*$, then $\nabla^2f(x^*) \succeq 0$. (Second Order Necessary Condition)
+\end{enumerate}
+\end{proposition}
+
+\begin{proof}[Proof of \propositionref{nc for local min}.]
+Fix any direction $d \in \R^n$.
+\begin{enumerate}
+    \item $g(\alpha) := f(x^* + \alpha d)$. Then
+    \begin{align}
+        0 &\le \lim_{\alpha \to 0} \frac{f(x^* + \alpha d) - f(x^*)}{\alpha} \label{eqn:0} \\
+        &= \frac{\partial g(0)}{\partial \alpha} \nonumber \\
+        &= d^\trans \nabla f(x^*) \nonumber
+    \end{align}
+    \ref{eqn:0} Because $x^*$ is a local minimum, $0 \le f(x^* + \alpha d) - f(x^*)$ for sufficiently small alpha. So, we can construct a sequence with only positive $\alpha$ that converges to $x^*$ such that each element $0 \le \frac{f(x^* + \alpha_n d) - f(x^*)}{\alpha_n}$ which implies that statement given that $f$ is locally differentiable.\\
+    \\
+    Since $d$ is arbitrary, this implies that $\nabla f(x^*) = 0$.
+    \item First we represent $f(x^* + \alpha d) - f(x^*)$ using the 2nd order Taylor expansion.
+    \begin{align*}
+        f(x^* + \alpha d) - f(x^*) &= \alpha \nabla f(x^*)^\trans d + \frac{\alpha^2}{2}d^\trans \nabla^2 f(x^*)d + O(\alpha^2) \nonumber\\
+        &= \frac{\alpha^2}{2} d^\trans \nabla^2 f(x^*) d + O(\alpha^2) \\
+    \end{align*}
+    Now we do the following
+    \begin{align*}
+        0 &\le \lim_{\alpha \to 0} \frac{f(x^* + \alpha d) - f(x^*)}{\alpha^2} \\
+        &= \lim_{\alpha \to 0} \frac{1}{2}d^\trans \nabla^2 f(x^*) d + \frac{O(\alpha^2)}{\alpha^2} \\
+        &= \frac{1}{2}d^\trans \nabla^2 f(x^*) d
+    \end{align*}
+    Because $d$ is arbitrary, this implies that $\nabla^2 f(x^*) \succeq 0$ (Positive semidefinite).
+    
+\end{enumerate}
+\end{proof}
+
+Note that $\nabla f(x^*) = 0$ alone does not imply $x^*$ is a local minimum. Even the necessary conditions $\nabla f(x^*) = 0$ and $\nabla^2 f(x^*) \succeq 0$ does not imply $x^*$ is a local minimum. This is because it could be that $\nabla^2 f(x^*) = 0$, but the 3rd order is not 0. For example in the 1d case, $x^* = 0$ for $f(x) = x^3$ satisfies these conditions, but is not a local minimum. Now, we will look at the actual sufficient conditions for a local minimum, but these conditions can only detect strict local minimum.
```

Comment:
`can only detect strict local minimum` -> `local minima`, since it's the plural here.